spa/src/components/pageEditor/editInterface/EditSaveControls.tsx

```diff
@@ -0,0 +1,39 @@
+import { Undo } from '@material-ui/icons';
+import PropTypes, { InferProps } from 'prop-types';
+import { Button, Root } from './EditSaveControls.styles';
+
+const EditSaveControlsPropTypes = {
+  cancelDisabled: PropTypes.bool,
+  onCancel: PropTypes.func.isRequired,
+  onUpdate: PropTypes.func.isRequired,
+  variant: PropTypes.oneOf(['cancel', 'undo']).isRequired
+};
+
+export interface EditSaveControlsProps extends InferProps<typeof EditSaveControlsPropTypes> {
+  onCancel: () => void;
+  onUpdate: () => void;
+  variant: 'cancel' | 'undo';
+}
+
+export function EditSaveControls({ cancelDisabled, onCancel, onUpdate, variant }: EditSaveControlsProps) {
+  return (
+    <Root>
+      {variant === 'cancel' && (
+        <Button color="secondary" disabled={!!cancelDisabled} onClick={onCancel}>
+          Cancel
+        </Button>
+      )}
+      {variant === 'undo' && (
+        <Button color="text" disabled={!!cancelDisabled} onClick={onCancel} startIcon={<Undo />}>
+          Undo
+        </Button>
+      )}
```

Comment:
I like using this pattern when we have a certainty that only these limited options will be available:
```
{
        {
          'foo': <Foo />,
          'bar': <Bar />
        }[param]
 }
```

```suggestion
      {{
         'cancel': (
         	<Button color="secondary" disabled={!!cancelDisabled} onClick={onCancel}>
             		Cancel
           	</Button>
           )
        'undo': (
        	<Button color="text" disabled={!!cancelDisabled} onClick={onCancel} startIcon={<Undo />}>
          		Undo
        	</Button>
      	)
      }[variant]}
```

Reply:
I like this suggestion--I rearranged it a little bit so that the lookup isn't in the render itself, let me know what you think: e7906a289fde922823035e9fde95c6fc88d2e08f